src/node/internal/internal_http_incoming.ts

```diff
@@ -0,0 +1,126 @@
+  // @ts-expect-error TS2416 Type-inconsistencies
+  public method: string | null = null;
+  // @ts-expect-error TS2416 Type-inconsistencies
+  public statusCode: number | null = null;
+  // @ts-expect-error TS2416 Type-inconsistencies
+  public statusMessage: string | null = null;
+  public httpVersionMajor = 1;
+  public httpVersionMinor = 1;
+  public httpVersion: string = '1.1';
+
+  // TODO(soon): Get rid of the second argument.
+  public constructor(
+    response: Response,
+    resetTimers: (opts: { finished: boolean }) => void
+  ) {
+    if (!(response instanceof Response)) {
+      // TODO(soon): Fully support IncomingMessage constructor
+      throw new ERR_METHOD_NOT_IMPLEMENTED('IncomingMessage');
+    }
+    super({});
+
+    this.#resetTimers = resetTimers;
+    this.#response = response;
+
+    if (this._readableState) {
+      this._readableState.readingMore = true;
+    }
+
+    this.url = response.url;
+    this.statusCode = response.status;
+    this.statusMessage = response.statusText;
+
+    this.on('end', () => {
+      queueMicrotask(() => this.emit('close'));
+    });
+
+    this.#reader = this.#response.body?.getReader();
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.#tryRead();
+  }
+
+  async #tryRead(): Promise<void> {
+    if (!this.#reader || this.#reading || this.#aborted) return;
+
+    this.#reading = true;
+
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      while (true) {
+        const { done, value } = await this.#reader.read();
+        if (done) {
+          break;
+        }
+        this.push(value);
+      }
+    } catch (error) {
+      this.emit('error', error);
+    } finally {
+      this.#reading = false;
+      this.#resetTimers({ finished: true });
+      this.push(null);
+    }
+  }
+
+  public override _read(): void {
+    if (!this.#reading && !this.#aborted) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.#tryRead();
+    }
+  }
+
+  public override _destroy(
+    error: Error | null,
+    callback: (error?: Error | null) => void
+  ): void {
+    if (!this.#aborted) {
+      this.#aborted = true;
+      this.emit('aborted');
+    }
+    this.#reading = false;
+
+    queueMicrotask(() => {
+      callback(error);
+    });
+  }
+
+  public get headers(): Record<string, string> {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-explicit-any
+    return Object.fromEntries(this.#response.headers as any);
+  }
+
+  // @ts-expect-error TS2416 Type inconsistency
+  public get headersDistinct(): Record<string, string> {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-explicit-any
+    return Object.fromEntries(this.#response.headers as any);
+  }
+
+  // @ts-expect-error TS2416 Type inconsistency
+  public get trailers(): Record<string, unknown> {
+    // Not supported.
```

Comment:
It might make sense to throw unsupported here?

Reply:
This is a getter. If we throw an error it might be too much destruptive. Even on console.log(req), it will throw.